packages/dialog/src/DialogBase.ts

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    CSSResultArray,
+    html,
+    PropertyValues,
+    SpectrumElement,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import { property } from '@spectrum-web-components/base/src/decorators.js';
+
+import '@spectrum-web-components/underlay/sp-underlay.js';
+import '@spectrum-web-components/button/sp-button.js';
+
+import '../sp-dialog.js';
+import modalWrapperStyles from '@spectrum-web-components/modal/src/modal-wrapper.css.js';
+import modalStyles from '@spectrum-web-components/modal/src/modal.css.js';
+import { Dialog } from './Dialog.js';
+import { FocusVisiblePolyfillMixin } from '@spectrum-web-components/shared';
+import { firstFocusableIn } from '@spectrum-web-components/shared/src/first-focusable-in.js';
+
+/**
+ * @element sp-dialog-wrapper
+ *
+ * @slot - A Dialog element to display.
+ * @fires close - Announces that the dialog has been closed.
+ */
+export class DialogBase extends FocusVisiblePolyfillMixin(SpectrumElement) {
+    public static override get styles(): CSSResultArray {
+        return [modalWrapperStyles, modalStyles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public dismissable = false;
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: String, reflect: true })
+    public mode?: 'fullscreen' | 'fullscreenTakeover';
+
+    /**
+     * When set to true, fills screens smaller than 350px high and 400px wide with the full dialog.
+     */
+    @property({ type: Boolean })
+    public responsive = false;
+
+    private transitionPromise = Promise.resolve();
+
+    private resolveTransitionPromise = (): void => {
+        return;
+    };
+
+    @property({ type: Boolean })
+    public underlay = false;
+
+    protected get dialog(): Dialog {
+        return (
```

Comment:
How can you be sure this won't return undefined?

Reply:
Updated so that it will fallback to `this` and added Dev Mode warning as this amounts to the element being broken.

Reply:
Isn't the idea of adding DialogBase for extendability beyond the usage of `sp-dialog`?

Reply:
It expects a "dialog" in it's default implementation, this could be an `sp-dialog` or something else. When extended, like in `sp-dialog-wrapper` the "dialog" could be supplied internally, which would require that this value be overridden, or there could be a more nuanced query for the root element that acts as both the arbiter of keyboard focus within the "dialog" content as well as surfacing any extended `updateComplete` metrics before such interactions should be possible. I'm trying to get at this with the additional documentation [found here](https://github.com/adobe/spectrum-web-components/pull/2542/files#diff-fbb28bf07874448fd6e351194f7691091964a2905d1108c60b093e9180ea73c7R54-R56). Let me know if you see where I can expand or clarify this for you and future dialog builders.